crates/sui-framework/packages/move-stdlib/sources/vector.move

```diff
@@ -4,368 +4,371 @@
-    /// Otherwise, returns false.
-    public fun contains<Element>(v: &vector<Element>, e: &Element): bool {
-        let mut i = 0;
-        let len = v.length();
-        while (i < len) {
-            if (&v[i] == e) return true;
-            i = i + 1;
-        };
-        false
-    }
+/// Return true if `e` is in the vector `v`.
+/// Otherwise, returns false.
+public fun contains<Element>(v: &vector<Element>, e: &Element): bool {
+    let mut i = 0;
+    let len = v.length();
+    while (i < len) {
+        if (&v[i] == e) return true;
+        i = i + 1;
+    };
+    false
+}
 
-    /// Return `(true, i)` if `e` is in the vector `v` at index `i`.
-    /// Otherwise, returns `(false, 0)`.
-    public fun index_of<Element>(v: &vector<Element>, e: &Element): (bool, u64) {
-        let mut i = 0;
-        let len = v.length();
-        while (i < len) {
-            if (&v[i] == e) return (true, i);
-            i = i + 1;
-        };
-        (false, 0)
-    }
+/// Return `(true, i)` if `e` is in the vector `v` at index `i`.
+/// Otherwise, returns `(false, 0)`.
+public fun index_of<Element>(v: &vector<Element>, e: &Element): (bool, u64) {
+    let mut i = 0;
+    let len = v.length();
+    while (i < len) {
+        if (&v[i] == e) return (true, i);
+        i = i + 1;
+    };
+    (false, 0)
+}
 
-    /// Remove the `i`th element of the vector `v`, shifting all subsequent elements.
-    /// This is O(n) and preserves ordering of elements in the vector.
-    /// Aborts if `i` is out of bounds.
-    public fun remove<Element>(v: &mut vector<Element>, mut i: u64): Element {
-        let mut len = v.length();
-        // i out of bounds; abort
-        if (i >= len) abort EINDEX_OUT_OF_BOUNDS;
-
-        len = len - 1;
-        while (i < len) v.swap(i, { i = i + 1; i });
-        v.pop_back()
-    }
+/// Remove the `i`th element of the vector `v`, shifting all subsequent elements.
+/// This is O(n) and preserves ordering of elements in the vector.
+/// Aborts if `i` is out of bounds.
+public fun remove<Element>(v: &mut vector<Element>, mut i: u64): Element {
+    let mut len = v.length();
+    // i out of bounds; abort
+    if (i >= len) abort EINDEX_OUT_OF_BOUNDS;
 
-    /// Insert `e` at position `i` in the vector `v`.
-    /// If `i` is in bounds, this shifts the old `v[i]` and all subsequent elements to the right.
-    /// If `i == v.length()`, this adds `e` to the end of the vector.
-    /// This is O(n) and preserves ordering of elements in the vector.
-    /// Aborts if `i > v.length()`
-    public fun insert<Element>(v: &mut vector<Element>, e: Element, mut i: u64) {
-        let len = v.length();
-        // i too big abort
-        if (i > len) abort EINDEX_OUT_OF_BOUNDS;
-
-        v.push_back(e);
-        while (i < len) {
-            v.swap(i, len);
-            i = i + 1
-        }
-    }
+    len = len - 1;
+    while (i < len) v.swap(i, {
+            i = i + 1;
+            i
+        });
```

Comment:
Was this auto formatted? If so, its a good bug to fix IMO :)

Reply:
Ty!